### PR TITLE
feat: [SIW-742] Add `cose-js` sign and common utils

### PR DESCRIPTION
**Pull request by hevelius**
### Short description

This PR adds `cose-js` utils to get encryption on cbor. This PR adds only base cose-js utils methods. In another PR we refactor the `doSign` method to perform an encryption using `io-react-native-crypto` to access SE/TEE.

Note: The `cose-js` files are base on this repository https://github.com/erdtman/cose-js

#### List of Changes

- Add `cose` directory with sign and common utils
- Add `node-rsa` as dependency
- Add `elliptic` as dependency
- Update example app to test COSE sign on a mocked plain text
- Update temp function in `ProximityManager` to perform a COSE sign

### CODE REVIEW

File: src/proximity-manager/index.ts [hunk 1]
```diff
@@ -325,6 +326,30 @@ const ProximityManager = () => {
...
+            'hex'
+          ),
+        },
+      };
+      const buf = await sign.create(headers, message, signer);
+      console.log('Signed message: ' + buf.toString('hex'));
+    } catch (error) {
+      console.log(error);
+    }
+  };
```
Comment (balanza) [inline src/proximity-manager/index.ts:351]: **question:** ‏What does this function does? Why it returns nothing?

**Review by balanza — CHANGES_REQUESTED** [review]
**issue:** Curious choice to go for as a plain vanilla JS file. We usually don't do it for several reasons:
* types help you describe what the code does
* types force you to reason about the possible combination of inputs a function can support
* types force you to parse input data before passing it to functions (implying input validation)
* types help you to spot corner cases on the code (_"I didn't think this value could be null"_)

Can we go for a vanilla JS solution? Sure, as long as the need is well-motivate and the safety loss is mitigated (with plenty of unit tests and comments).

My advice is we either use Typescript (and we invest time in type-modeling) or we have tons of tests and comments.

‏

**Comment by hevelius** [comment]
> **issue:** Curious choice to go for as a plain vanilla JS file. We usually don't do it for several reasons:
> 
> * types help you describe what the code does
> * types force you to reason about the possible combination of inputs a function can support
> * types force you to parse input data before passing it to functions (implying input validation)
> * types help you to spot corner cases on the code (_"I didn't think this value could be null"_)
> 
> Can we go for a vanilla JS solution? Sure, as long as the need is well-motivate and the safety loss is mitigated (with plenty of unit tests and comments).
> 
> My advice is we either use Typescript (and we invest time in type-modeling) or we have tons of tests and comments.
> 
> ‏

@balanza the `cose-js` code is based on another repository and this is its original code. However it is not yet clear whether we can use this `cose-js` implementation for reasons related to the ISO-18013 standard. For velocity reasons, the transition to typescript is not envisaged. Refactoring in TypeScript will be the next step. I added a Jira task about it.

**Comment by balanza** [comment]
I don't understand the core problem. is there a technical reason we cannot use Typescript, or is just a consideration on the effort?

**Comment by hevelius** [comment]
> I don't understand the core problem. is there a technical reason we cannot use Typescript, or is just a consideration on the effort?

Related to Typescript only effort

**Comment by balanza** [comment]
> Related to Typescript only effort

Ok, thanks. I stick with my opinion: a JS implementation with a comparable level of _"quality"_ requires more accuracy in testing and documenting. We can compare this extra effort with the type-gymnastic we are saving. 

Otherwise, we are deliberately adding some below-the-standard code, I don't think we need to do it.

File: src/proximity-manager/index.ts (hunk 1 above)
Comment (grausof) [inline src/proximity-manager/index.ts:351]: What do you think about making this function more explicit, perhaps having the header also taken as parameters and returning the signed buffer?

**Review by grausof — APPROVED** [review]
LGTM

**Comment by grausof** [comment]
> > Related to Typescript only effort
> 
> Ok, thanks. I stick with my opinion: a JS implementation with a comparable level of _"quality"_ requires more accuracy in testing and documenting. We can compare this extra effort with the type-gymnastic we are saving.
> 
> Otherwise, we are deliberately adding some below-the-standard code, I don't think we need to do it.

@balanza @hevelius Take a look here https://github.com/erdtman/cose-js/pull/45

File: src/proximity-manager/index.ts (hunk 1 above)
Comment (hevelius) [reply src/proximity-manager/index.ts:351]: @grausof Yes. If you agree I would do it in this other branch/PR. What do you think? This PR only add the cose-js utils.

Comment (grausof) [reply src/proximity-manager/index.ts:351]: OK 👌

**Review by balanza — APPROVED** [review]
